packages/node-core/src/indexer/fetch.service.ts

```diff
@@ -23,6 +23,7 @@ const DICTIONARY_MAX_QUERY_SIZE = 10000;
 const CHECK_MEMORY_INTERVAL = 60000;
 
 export abstract class BaseFetchService<
+  API extends ApiService,
```

Comment:
I think this can be, if we use the interface then it simplifies writing tests
```suggestion
  API extends IApi,
```